src/message/messageActionSheet.js

```diff
@@ -82,6 +86,21 @@ const reply = ({ message, dispatch, ownUser }) => {
 reply.title = 'Reply';
 reply.errorMessage = 'Failed to reply';
 
+const replyWithMention = ({ message, dispatch, ownUser, narrow, startReplyWithMention }) => {
+  const replyNarrow = getNarrowForReply(message, ownUser.user_id);
+  const replyMessage = `@**${message.sender_full_name}**`;
+  if (JSON.stringify(replyNarrow) !== JSON.stringify(narrow)) {
+    dispatch(draftUpdate(replyNarrow, replyMessage));
```

Comment:
Is there a reason `draftUpdate` doesn't work for both of these cases?

Reply:
`draftUpdate` actions work when `ComposeBox` first gets rendered, so if we are at the same narrow, the `draftUpdate` action will fire, but the `ComposeBox` will not be updated it gets re-rendered.

Reply:
Actually, Greg answered that question for me, [here is the answer](https://chat.zulip.org/#narrow/stream/48-mobile/topic/Reply.20with.20mention/near/1160754)